make use of most common content locales

mirceachira opened the pull request:
closes #714

Gallaecio commented:
Hmm… I don’t see how these changes could cause those CI issues :confused:

noviluni commented:
Hi @mirceachira, @Gallaecio 

When running `python order_languages.py` (As @mirceachira  did), we get the order from `https://github.com/unicode-cldr/cldr-dates-full.git`.

Those errors in the pipelines are happening because CLDR included new languages, and when we try to import the languages from the list of languages ( `dateparser.data.language_info.language_order`) we can't find them.

I think the new locales are: ccp, ceb, ff-Latn, ia, jv, ku, mi, sd, tg, tt, wo, xh, yue-Hans and yue-Hant.

To create the files for the new languages we should run:

1. `python get_cldr_data.py` It creates the json files
2. `python write_complete_data.py` It creates the py files.


However, we should also:
* Update the "supported locales" list in the docs
* Update the `avoid_languages` list from `dateparser.dateparser_scripts`. We may need to add some of the new languages or remove some of the old languages from this list.
* Run again the `order_languages` script as it should reflect the `avoid_languages` changes.

@mirceachira would you like to work on this? I think you could create a separate PR just running the three scripts (and updating the docs and the `avoid_languages` list). Once we merge that, we can come here and update this PR and we won't see any issue. 

Of course, if you don't want/ you can't I can do it. :)

mirceachira commented:
Hey @noviluni for sure I would like to pick that up! Thanks for the detailed tips on it!

noviluni commented:
Hi @mirceachira, you will be able run `python dateparser_scripts/order_languages.py` without errors after merging this: https://github.com/scrapinghub/dateparser/pull/825

However, I checked what you did and I think this is wrong: https://github.com/scrapinghub/dateparser/pull/805/files#diff-1c848ec911779bd8e06fea6e61745c57c43a463bab86d8ef955f2e29ec8484d0R60 because is generating multiple times the same language in the `language_order` list in `languages_info.py`. 

On the other hand, I would really appreciate it if you add the date when that list was created in this comment: https://github.com/scrapinghub/dateparser/pull/805/files#diff-1c848ec911779bd8e06fea6e61745c57c43a463bab86d8ef955f2e29ec8484d0R15 to allow us to update in in the future.

noviluni commented:
Hi @mirceachira, we merged the other PR (#825), could you rebase this PR against master? :slightly_smiling_face:

mirceachira commented:
@noviluni simply running the script generates the same problem you mentioned here https://github.com/scrapinghub/dateparser/pull/805#issuecomment-705368883

Though there are much fewer tests that required fixing now after #825 . I added the languages accordingly to 'avoid_languages' and fixed the remaining issues. 

Please note that I removed this test: https://github.com/scrapinghub/dateparser/pull/805/commits/11ba3fbf10576b1972b70ba41a6995f87994b4f4
I'm not sure why the locale order resulted in its failure as well as the failure in the shortname test that I fixed here https://github.com/scrapinghub/dateparser/pull/805/commits/876941eb31687602d7d5e5985e37950730524920

Gallaecio commented on dateparser_scripts/order_languages.py:
```diff
@@ -9,7 +9,10 @@
+avoid_languages = {
+    'ia', 'mi', 'ccp', 'tt', 'yue-Hant', 'ku', 'ff-Latn', 'tg', 'xh', 'wo', 'ceb', 'sd', 'yue-Hans',
+    'jv', 'cu', 'kkj', 'nds', 'prg', 'tk', 'vai', 'vai-Latn', 'vai-Vaii', 'vo'
+}
```
:lipstick: Can we keep this list alphabetical?

Gallaecio commented on dateparser_scripts/write_complete_data.py:
```diff
@@ -7,6 +7,7 @@
 from ruamel.yaml import RoundTripLoader
 
 from dateparser_scripts.utils import combine_dicts
+from dateparser_scripts.order_languages import avoid_languages
```
:lipstick: 

```suggestion
from dateparser_scripts.order_languages import avoid_languages
from dateparser_scripts.utils import combine_dicts
```

noviluni commented on dateparser_scripts/order_languages.py:
```diff
@@ -9,7 +9,19 @@
 
 # Languages with insufficient translation data are excluded
-avoid_languages = ['cu', 'kkj', 'nds', 'prg', 'tk', 'vai', 'vai-Latn', 'vai-Vaii', 'vo']
+avoid_languages = {
+    'ccp', 'ceb', 'cu', 'ff-Latn', 'ia', 'jv', 'kkj', 'ku', 'mi', 'nds', 'prg',
+    'sd', 'tg', 'tk', 'tt', 'vai', 'vai-Latn', 'vai-Vaii', 'vo', 'wo', 'xh',
+    'yue-Hans', 'yue-Hant'
+}
```
You wrote here some languages from the next CLDR versions. Could you revert this?

```suggestion
avoid_languages = {'cu', 'kkj', 'nds', 'prg', 'tk', 'vai', 'vai-Latn', 'vai-Vaii', 'vo'}
```

noviluni reviewed (changes requested):
Hi @mirceachira! 

I think we are almost ready to merge this! I'm ok with the changes in the tests, but I would need you to revert the `avoid_languages` set as suggested and run this: `python dateparser_scripts/order_languages.py` again.

After that I think we will be able to merge it! :rocket:

noviluni commented on dateparser_scripts/order_languages.py:
```diff
@@ -9,7 +9,15 @@
+
+# Order from https://w3techs.com/technologies/overview/content_language
+# Last updated on 30.10.2020
+most_common_locales = [
+    'en', 'ru', 'es', 'tr', 'fa', 'fr', 'de', 'ja', 'pt', 'vi', 'zh', 'ar', 'it', 'pl', 'in', 'el',
+    'nl', 'ko', 'th', 'iw', 'uk', 'cs', 'sv', 'ro', 'hu', 'da', 'sr', 'sk', 'fi', 'bg', 'hr', 'lt',
+    'hi', 'no_NO', 'sl', 'no', 'et', 'lv'
+]
```
Hi @mirceachira, I rechecked this list before merging and it seems that "Hebrew", "Norwegian Bokmål" and "Norwegian (Nynorsk)" were wrong (those codes you wrote are used in other specifications, but they are not in the `language_order` list). Could you change it? Thanks!

```suggestion
most_common_locales = [
    'en', 'ru', 'es', 'tr', 'fa', 'fr', 'de', 'ja', 'pt', 'vi', 'zh', 'ar', 'it', 'pl', 'in', 'el',
    'nl', 'ko', 'th', 'he', 'uk', 'cs', 'sv', 'ro', 'hu', 'da', 'sr', 'sk', 'fi', 'bg', 'hr', 'lt',
    'hi', 'nb', 'sl', 'nn', 'et', 'lv'
]
```

noviluni replied:
(I have just seen that you mentioned them as "new" in this comment: https://github.com/scrapinghub/dateparser/issues/714#issuecomment-701373790, but I didn't see that before, sorry)

noviluni commented:
I've been opening and closing this PR because we just switched from travis-ci.org to travis-ci.com but the checks show the old travis-ci.org...

Anyway, it seems that all tests passed in travis-ci.com: https://travis-ci.com/github/scrapinghub/dateparser/builds/204189662

noviluni commented:
Ok, this is a hard PR to review :sweat_smile: 

You should change the "in" code by "id" in the `most_common_locales` list too, and after that you should run `python dateparser_scripts/write_complete_data.py`.

noviluni commented:
Hi @mirceachira, thanks for fixing this. As travis-ci stopped working we can't run the tests, so I can't merge this. I started migrating to `github actions` (WIP PR: https://github.com/scrapinghub/dateparser/pull/859/files) after that we will be able to run the tests and merge this :)

noviluni reviewed (approved):
LGTM!
